### PR TITLE
Programming Languages - "AngularJs" Complete Resources

### LINKED ISSUE #456: 💡 [FEATURE] - Add "AngularJS" Complete Resources
### Idea Contribution

- [X] I have read all the feature request issues.
- [X] I'm interested in working on this issue
- [X] I'm part of GSSOC organization

### Explain feature request

AngularJS, a robust framework for building dynamic web applications, requires comprehensive documentation to support developers in creating sophisticated client-side applications.

### Explain your solution

I propose adding a dedicated section for AngularJS resources in the Programming Languages documentation,

Contents of AngularJS/readme.md:

- AngularJS Basics:

    - Setting up the AngularJS environment
    - Understanding the MVC architecture
    - Core concepts and directives
- Advanced AngularJS:

    - Dependency injection
    - Building single-page applications (SPAs)
    - Advanced directives and components
- AngularJS Tools and Libraries:

    - Using the AngularJS CLI
    - Popular modules and libraries (UI-Router, ngResource)
    - Testing with Jasmine and Karma
- Performance and Optimization:
    - Optimizing AngularJS applications
    - Best practices for large applications

- Deployment:
    - Deployment strategies for AngularJS applications
    - Using Docker with AngularJS
    - Continuous integration and delivery
- AngularJS Projects:
   - Real-world applications

### Any alternative approaches/features

To enhance the learning experience, include essential resources such as recommended books, YouTube channels, and websites. Additionally, providing a beginner-friendly roadmap can help new developers navigate through the resources step-by-step.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Santhosh-Siddhardha**
**IMPORTANT:** When creating your Pull Request, please use the following format for the title: `Category - Resource Name & Section`  

PR Title samples:

- `Web development - CSS Portal & Tools`
- `Web development - Leetcode & Challenges`

Closes #456 

### Description: 
Hey @jfmartinz I have added the resources of AngularJs

```
 ./Programming Languages
    /AngularJs
     /readme.md
```
Please take a look, Thank you.

### Checklist:
Before submitting your pull request, ensure that you have completed the following tasks: 

- [x] I have carefully reviewed and adhered to the [contributing guidelines](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md) before creating this pull request.
- [x] I followed the prescribed PR title template. _(Check this if you're adding a resource)_

> fill the bracket with x. Something like this [x]

### CODE REVIEW

File: Programming Languages/AngularJS/readme.md [hunk 1]
```diff
@@ -0,0 +1,1084 @@
...
+   - [State Management](#state-management)
+   - [Dynamic Component Loading](#dynamic-component-loading)
+   - [Angular Ivy](#angular-ivy)
+- [AngularJS Projects](#angularjs-projects)
+   - [Beginner Level Projects](#beginner-level-projects)
+   - [Intermediate Level Projects](#intermediate-level-projects)
+   - [Advanced Level Projects](#advanced-level-projects)
+- [Version Control](#version-control)
+- [YouTube](#youtube)
+   - [YouTube Channels to Learn AngularJS](#youtube-channels-to-learn-angularjs)
```
Comment (jfmartinz) [inline Programming Languages/AngularJS/readme.md:52]:
```suggestion
   - [YouTube Channels](#youtube-channels)
```

**Review by jfmartinz — CHANGES_REQUESTED** [review]

**Comment by Santhosh-Siddhardha** [comment]
@jfmartinz Made changes, Have a look

**Comment by jfmartinz** [comment]
No need. I think

**Comment by Santhosh-Siddhardha** [comment]
@jfmartinz Today all the Contributors got a mail, and these important things were mentioned in that mail, Have a look

**Comment by jfmartinz** [comment]
Only in 'Issue', I think no need in PR. You just need to reference the issue, let me confirm it

**Comment by Santhosh-Siddhardha** [comment]
Once have a look in this , https://github.com/GSSoC24/Contributor/discussions/288?utm_source=substack&utm_medium=email

**Comment by jfmartinz** [comment]
Okay, sure. THey announced too late again, ive merged alot prs already.

**Comment by jfmartinz** [comment]
I will do that.

**Comment by Santhosh-Siddhardha** [comment]
@jfmartinz thank you

**Comment by jfmartinz** [comment]
Can you please not mentioned again, I will jusst assigned and add labels to the PRs merged. My notif is flooded right now, hope you understand. Just remind me if i forgot

**Comment by Santhosh-Siddhardha** [comment]
Yeah sure.... I can understand

**Comment by sanjay-kv** [comment]
@Santhosh-Siddhardha as per guidelines only 200 points are allowed. 
image

**Pull request merged**